src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -408,7 +409,13 @@ public int iota() {
         return iota.incrementAndGet();
     }
 
-    int currentNodeCount() {
+    /**
+     * Returns an approximate size of the flow graph, based on the heuristic that the iota is incremented once per new node.
+     * The exact count may be a little different due to special cases.
+     * ({@link FlowNodeStorage} does not currently offer a size, or a set of all nodes.
+     * An exact count could be obtained with {@link FlowGraphWalker}, but this could be more overhead.)
```

Comment:
Much, much more overhead, since you're creating a HashSet of all existent flownodes. ;)

Multiple seconds even is possible.